### PR TITLE
Add ChatTLFResolve notification

### PR DESCRIPTION
**Pull request by mlsteele**
This `ChatTLFResolve` notification is sent to electron when a client gets a `chat.tlfresolve` `OOBM`. This notification comes along when a conversation you were already in has someone SBS resolve in it.

cc @chrisnojima

### CODE REVIEW

File: go/libkb/notify_router.go [hunk 1]
```diff
@@ -556,6 +557,34 @@ func (n *NotifyRouter) HandleChatTLFFinalize(ctx context.Context, uid keybase1.U
 	n.G().Log.Debug("- Sent ChatTLFFinalize notfication")
 }
 
+func (n *NotifyRouter) HandleChatTLFResolve(ctx context.Context, uid keybase1.UID, convID chat1.ConversationID) {
+	if n == nil {
+		return
+	}
+	var wg sync.WaitGroup
+	n.G().Log.Debug("+ Sending ChatTLFResolve notfication")
```
Comment (mmaxim) [inline go/libkb/notify_router.go]: Typo

File: go/protocol/chat1/notify.go [hunk 2]
```diff
@@ -215,6 +215,11 @@ type ChatTLFFinalizeArg struct {
 	FinalizeInfo ConversationFinalizeInfo `codec:"finalizeInfo" json:"finalizeInfo"`
 }
 
+type ChatTLFResolveArg struct {
+	Uid    keybase1.UID   `codec:"uid" json:"uid"`
+	ConvID ConversationID `codec:"convID" json:"convID"`
```
Comment (mmaxim) [inline go/protocol/chat1/notify.go]: Shouldn't we be sending the new TLF name?

Comment (mlsteele) [inline go/protocol/chat1/notify.go]: Probably, but we don't have that from kbfstlfd yet. Will almost definitely add this when attacking sbs conflicts. For now the frontend could just refresh and the existing naming stuff takes care of it I think.

Comment (mmaxim) [inline go/protocol/chat1/notify.go]: The client could just get the name though

Comment (mmaxim) [inline go/protocol/chat1/notify.go]: I just don't think this notification has any value unless we pass up the new name, what other information would the UI care about here?

Comment (mlsteele) [inline go/protocol/chat1/notify.go]: They would care about the fact that it happened, so they can refresh the stale name. But you're right the service can get it, I'll add that,

**Comment by mlsteele** [comment]
Added `ConversationResolveInfo` which has the new `tlfName` as calculated by the service. I expect this will grow when fixing conflicts.

File: go/service/gregor.go [hunk 3]
```diff
@@ -942,6 +944,46 @@ func (g *gregorHandler) chatTlfFinalize(ctx context.Context, m gregor.OutOfBandM
...
+	dec := codec.NewDecoder(reader, &codec.MsgpackHandle{WriteExt: true})
+	err := dec.Decode(&update)
+	if err != nil {
+		return err
+	}
+
+	uid := m.UID().String()
+
+	// Get and localize the conversation to get the new tlfname.
+	inbox, _, err := g.G().InboxSource.ReadNoCache(ctx, m.UID().Bytes(), nil, &chat1.GetInboxLocalQuery{
```
Comment (mlsteele) [inline go/service/gregor.go]: Can `InboxSource.Read` be used here to hit the cache? Or does that only work with some queries? Or would it cause the tlfname not to be re-resolved?

Also, is this slow enough or have the potential to hang and if so should put this notifyrouter send in a goroutine?

Comment (mlsteele) [inline go/service/gregor.go]: Ok, tested and done.

**Pull request merged**